lightning/backend/local.go

```diff
@@ -1181,10 +1181,14 @@ func (local *local) ImportEngine(ctx context.Context, engineUUID uuid.UUID) erro
 		log.L().Info("start import engine", zap.Stringer("uuid", engineUUID),
 			zap.Int("ranges", len(ranges)))
 
+		// if all the kv can fit in one region, skip split regions. TiDB will split one region for
+		// the table when table is created.
+		needSplit := len(ranges) > 0 || lf.TotalSize > local.regionSplitSize || lf.Length > regionMaxKeyCount
```

Comment:
is `needSplit := len(ranges) > 1` enough? seems `readAndSplitIntoRange` has checked `regionSplitSize` and `regionMaxKeyCount`

https://github.com/pingcap/tidb-lightning/blob/d9254b1deea8607f6ffd0f94a9cae3135d272745/lightning/backend/local.go#L811-L815

and `splitRangeBySizeProps` uses a tighter `keysLimit` so it will return more than 1 `ranges`

Reply:
splitRangeBySizeProps may still generate ranges that exceed regionSplitSize, so if the split is triggered by remain ranges, it may be a single range but the range size is exceeded.

Reply:
I see. Now I suggest

```suggestion
		needSplit := len(ranges) > 1 || lf.TotalSize > local.regionSplitSize || lf.Length > regionMaxKeyCount
```

Reply:
done